develop/tutorials/articles/03-portlets/packaging-a-jsf-application.markdown

```diff
@@ -198,18 +191,22 @@ approaches can be done by using the JSF Expression Language (EL):
         this.userService = userService;
     }
 
-Then you can write an
+Or you could more directly use the facesContext with the Liferay Faces Bridge as follows:
+
+	UserLocalService userLocalService = (UserLocalService) facesContext.getExternalContext()
+						.getApplicationMap().get("userLocalService");
```

Comment:
This approach will only work if the class that extends GenericFacesPortlet (annotated with @Component) saves-off the reference to the injected service as a PortletContext attribute. See PrimeFacesUsersPortlet.java for an example.